### PR TITLE
fix: transfer settings store data during GC

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

@LesnyRumcajs let me know if I'm wrong but I did not find the logic that transfers settings to the new database during garbage collection in https://github.com/ChainSafe/forest/pull/3320. This PR tries to fix the issue

Changes introduced in this pull request:

- transfer settings store data during GC

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — CHANGES_REQUESTED** [review]
Keeping a static list of all settings keys is not a robust solution. Let's sunset the GC instead.

**Comment by hanabi1224** [comment]
> Keeping a static list of all settings keys is not a robust solution. Let's sunset the GC instead.

@lemmih  As discussed on Slack, changed to enumerate DB keys instead of keeping a static list

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
It might be temporary, but let's have some tests for this.

File: src/db/rolling/impls.rs [hunk 1]
```diff
@@ -238,14 +264,39 @@ fn delete_db(db_path: &Path) {
...
     use rand::Rng;
     use tempfile::TempDir;
 
     use super::*;
+    use crate::libp2p_bitswap::BitswapStoreRead;
+
+    #[quickcheck]
+    fn ensure_settings_are_transferred(keys: Vec<String>) -> Result<()> {
+        let db_root = TempDir::new()?;
+        println!("Creating rolling db under {}", db_root.path().display());
```
Comment (lemmih) [inline src/db/rolling/impls.rs:281]: This debug println can be removed.

**Review by lemmih — CHANGES_REQUESTED** [review]
Updating `Cargo.lock` doesn't appear to be necessary. Let's leave those changes out.

File: src/db/rolling/impls.rs (hunk 1 above)
Comment (hanabi1224) [reply src/db/rolling/impls.rs:281]: Removed

**Comment by hanabi1224** [comment]
> Updating `Cargo.lock` doesn't appear to be necessary. Let's leave those changes out.

@lemmih removed

**Review by lemmih — APPROVED** [review]

**Pull request merged**